ip_country/src/country_finder.rs

```diff
@@ -225,77 +191,84 @@ mod tests {
     }
 
     #[test]
-    fn real_test_ipv4_with_sk_isp() {
-        let timestart = SystemTime::now();
+    fn real_test_ipv4_with_sk_ip() {
+        let _ = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("213.81.185.100").unwrap(), // dig www.zoznam.sk A
+        )
+        .unwrap();
+        let time_start = SystemTime::now();
         let result = CountryCodeFinder::find_country(
             &COUNTRY_CODE_FINDER,
             IpAddr::from_str("213.81.185.100").unwrap(), // dig www.zoznam.sk A
         )
         .unwrap();
-        let timeend = SystemTime::now();
+        let time_end = SystemTime::now();
 
         assert_eq!(result.free_world, true);
         assert_eq!(result.iso3166, "SK".to_string());
         assert_eq!(result.name, "Slovakia".to_string());
-        let duration = timeend.duration_since(timestart).unwrap().as_secs();
-        println!("duration: {}", duration);
-        assert!(duration <= 5);
+        let duration = time_end.duration_since(time_start).unwrap();
+        assert!(
+            duration.as_secs() < 1,
+            "Duration of the search was too long: {} ms",
+            duration.as_millis()
+        );
     }
 
     #[test]
     fn real_test_ipv6_with_google() {
-        let timestart = SystemTime::now();
+        let _ = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("2607:f8b0:4009:814::2004").unwrap(), // dig www.google.com AAAA
+        )
+        .unwrap();
+        let time_start = SystemTime::now();
         let result = CountryCodeFinder::find_country(
             &COUNTRY_CODE_FINDER,
             IpAddr::from_str("2607:f8b0:4009:814::2004").unwrap(), // dig www.google.com AAAA
         )
         .unwrap();
-        let timeend = SystemTime::now();
+        let time_end = SystemTime::now();
 
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "US".to_string());
         assert_eq!(result.name, "United States".to_string());
-        let duration = timeend.duration_since(timestart).unwrap().as_secs();
-        println!("duration: {}", duration);
-        assert!(duration <= 5);
+        let duration = time_end.duration_since(time_start).unwrap();
+        assert!(
+            duration.as_secs() < 1,
+            "Duration of the search was too long: {} ms",
+            duration.as_millis()
+        );
     }
 
     #[test]
     fn deserialize_country_blocks_ipv4_and_ipv6_adn_fill_vecs() {
```

Comment:
This whole test should be performed probably in short by simply testing the constructor for CountryCodeFinder, because it's a function that contains exactly the same logic which you're  strangely testing here separately. Seems clumsy to me.

Reply:
but in that way we will be unable to track those parts of the deserialization separately, as was Dan's proposal.

Reply:
You don't test them separately anyway so what are you complaining about? 

If you want to test them separately to be able to put forth how much time is needed for IpV4 and how much for IpV6, do as you please, it's also possible. I was just commenting on that your test represented exactly what is inside the constructor. If you'd shown an intention to test them separately I'd probably not have suggested such a change because it would've lacked sense. True.

Reply:
I tested separately deserialization and filling the vectors. That was suggestion from Dan and I saw it reasonable. If  want to suggest some more assertion, please let me know what do you see appropriate, because I can't see the benefit of asserting on separate deserialization of ipv4 and ipv6 country blocks, because these can vary and it does not bother us too much. We want to know how efficient is process of whole deserialization vs flling the vecs, because it affects the loading time of the Node. And we will be allways deserializing both version.

The reason, why we track down the time for deserialization and filling the vectors, in separate duration timestamps is, that if the filling of vetors could rise to some considerable duration, we can optimize this process by constructing enough memory space right on the compilation time.